trin-bridge/src/bridge.rs

```diff
@@ -82,13 +100,14 @@ impl Bridge {
                 let futures = futures::stream::iter(gossip_range.clone().map(|height| {
                     let gossip_stats = gossip_stats.clone();
                     async move {
-                        let _ = Bridge::serve_full_block(
-                            height,
-                            None,
-                            gossip_stats,
-                            self.portal_clients.clone(),
-                        )
-                        .await;
+                        let _ = self
```

Comment:
When I first implemented this functionality, I specifically tried to avoid using a shared reference, since I was using `tokio::spawn` and didn't want to have a shared reference across threads. Now, to be honest, I'm not sure what effect this actually has on performance, but my assumption was that we will eventually want to have fairly high concurrency in the bridge, so it's better to avoid shared refs across threads. I do agree that the shared ref does help simplify the code, but wonder if you have any opinions on the performance side of things?

Reply:
I see your concerns. I'm not sure what concurrency we will need and the future is unknown to me. I am more of an advocate of simple usage right now and adding complexity when the need arises.